### PR TITLE
Document Nakadi Publishing failure handling

### PR DESCRIPTION
**Pull request by otrosien**
Adds @ePaul 's nice assessment of the status quo to the API docs.

Relates to #415

### CODE REVIEW

**Comment by adyach** [comment]
@otrosien could we also add it to README as a separate paragraph?

**Comment by otrosien** [comment]
@adyach done.

**Review by rob-h-w — APPROVED** [review]
LGTM

**Comment by adyach** [comment]
👍

**Comment by otrosien** [comment]
👍

File: README.md [hunk 1]
```diff
@@ -189,6 +189,29 @@ Fahrschein supports different exponential backoff strategies when streaming even
...
+to handle exceptions when publishing is to create a retry-wrapper around the `NakadiClient.publish` method.
+
+In case of a partial success or also in cases like validation errors, which are complete failures, Fahrschein
+will throw an `EventPublishingException` with the `BatchItemResponse`s (as returned from Nakadi) for the failed
+ items in the responses property.
+
+These objects have the eid of the failed event, a `publishingStatus` (failed/aborted/submitted - but successful itemes are
+filtered out), the step where it failed and a detail string.
+
+If the application sets the eids itself (i.e. doesn't let Nakadi do it) and keeps track of them, this allows it
```
Comment (a1exsh) [inline README.md:204]: Nakadi does not assign `eid` to events — the id has to be provided by the event publisher.

Comment (adyach) [reply README.md:204]: https://github.com/zalando-nakadi/fahrschein/pull/422

Comment (ePaul) [reply README.md:204]: Huh, that's new to me. From my experience, when no `eid` is in the `metadata`, Nakadi will provide a random UUID. Did this change?

Comment (adyach) [reply README.md:204]: occurred_at and eid are required fields. Nakadi enriches only those https://github.com/zalando/nakadi/blob/026476faf60e8294d206f014b4c667913fb34106/core-common/src/main/java/org/zalando/nakadi/enrichment/MetadataEnrichmentStrategy.java#L37-L42

Comment (ePaul) [reply README.md:204]: I just tried it, and events without `eid` are rejected. I guess I mixed this up with some other metadata fields. Sorry.

**Pull request merged**